File: src/main/java/com/moovie/backend/domain/usecase/movies/GetUserFavoriteMovies.java
package com.moovie.backend.domain.usecase.movies;

import com.moovie.backend.domain.repository.MoviesRepository;
import com.moovie.backend.domain.model.MovieListing;
import org.springframework.stereotype.Service;

@Service
public class GetUserFavoriteMovies {

    private final MoviesRepository moviesRepository;

    public GetUserFavoriteMovies(MoviesRepository moviesRepository) {
        this.moviesRepository = moviesRepository;
    }

    public MovieListing invoke(String userId, int page) {
        return moviesRepository.getUserFavoriteMovies(userId, page);
    }
}

File: src/main/kotlin/org/example/domain/usecase/movies/GetUserFavoriteMovies.kt
package org.example.domain.usecase.movies

import kotlinx.coroutines.runBlocking
import org.example.domain.repository.MoviesRepository
import org.example.model.MovieListing

class GetUserFavoriteMovies(private val repository: MoviesRepository) {
    operator fun invoke(userId: String, page: Int): MovieListing = runBlocking {
        repository.getUserFavoriteMovies(userId, page)
    }
}
